### PR TITLE
[3.0] Remove most legacy tests

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

An "easy" but huge PR, focussing on removing legacy tests and only that.

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Tests/Form/ChoiceList/ORMQueryBuilderLoaderTest.php [hunk 1]
```diff
@@ -17,25 +17,6 @@
 
 class ORMQueryBuilderLoaderTest extends \PHPUnit_Framework_TestCase
 {
-    /**
-     * @expectedException \Symfony\Component\Form\Exception\UnexpectedTypeException
-     */
-    public function testItOnlyWorksWithQueryBuilderOrClosure()
-    {
-        new ORMQueryBuilderLoader(new \stdClass());
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/Tests/Form/ChoiceList/ORMQueryBuilderLoaderTest.php]: ORMQueryBuilderLoader is going to be type hinted in 3.0 so this test is legacy also

File: src/Symfony/Component/Form/Tests/ChoiceList/Factory/CachingFactoryDecoratorTest.php [hunk 2]
```diff
@@ -155,127 +155,6 @@ public function testCreateFromChoicesDifferentValueClosure()
         $this->assertSame($list2, $this->factory->createListFromChoices($choices, $closure2));
     }
 
-    public function testCreateFromFlippedChoicesEmpty()
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Tests/ChoiceList/Factory/CachingFactoryDecoratorTest.php]: createFromFlippedChoices is deprecated but doesn't trigger any deprecation (fixed in an other PR)

**Comment by Tobion** [comment]
Legacy Form types in the fixtures must be removed as well.

Status: Needs Work

**Comment by nicolas-grekas** [comment]
More fixtures and legacy tests removed.
Status: needs review

**Comment by Tobion** [comment]
:+1: 
Status: Reviewed

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**